src/zenml/integrations/kserve/steps/kserve_deployer.py

```diff
@@ -278,3 +317,154 @@ def kserve_model_deployer_step(
+    Raises:
+        ValueError: if the custom deployer parameters is not defined
+        DoesNotExistException: if no active stack is found
+
+
+    Returns:
+        KServe deployment service
+    """
+    # verify that a custom deployer is defined
+    if not config.custom_deploy_parameters:
+        raise ValueError(
+            "Custom deploy parameter which contains the path of the",
+            "custom predict function is required for custom model deployment.",
+        )
+
+    # get the active model deployer
+    model_deployer = KServeModelDeployer.get_active_model_deployer()
+
+    # get pipeline name, step name, run id and pipeline requirements
+    step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
+    pipeline_name = step_env.pipeline_name
+    pipeline_run_id = step_env.pipeline_run_id
+    step_name = step_env.step_name
+    docker_configuration = step_env.docker_configuration
+    runtime_configuration = step_env.runtime_configuration
+
+    # update the step configuration with the real pipeline runtime information
+    config.service_config.pipeline_name = pipeline_name
+    config.service_config.pipeline_run_id = pipeline_run_id
+    config.service_config.pipeline_step_name = step_name
+
+    # fetch existing services with same pipeline name, step name and
+    # model name
+    existing_services = model_deployer.find_model_server(
+        pipeline_name=pipeline_name,
+        pipeline_step_name=step_name,
+        model_name=config.service_config.model_name,
+    )
+
+    # even when the deploy decision is negative if an existing model server
+    # is not running for this pipeline/step, we still have to serve the
+    # current model, to ensure that a model server is available at all times
+    if not deploy_decision and existing_services:
+        logger.info(
+            f"Skipping model deployment because the model quality does not "
+            f"meet the criteria. Reusing the last model server deployed by step "
+            f"'{step_name}' and pipeline '{pipeline_name}' for model "
+            f"'{config.service_config.model_name}'..."
+        )
+        service = cast(KServeDeploymentService, existing_services[0])
+        # even when the deploy decision is negative, we still need to start
+        # the previous model server if it is no longer running, to ensure that
+        # a model server is available at all times
+        if not service.is_running:
+            service.start(timeout=config.timeout)
+        return service
+
+    # entrypoint for starting KServe server deployment for custom model
+    entrypoint_command = [
+        "python",
+        "-m",
+        "zenml.integrations.kserve.custom_deployer.zenml_custom_model",
+        "--model_name",
+        config.service_config.model_name,
+        "--predict_func",
+        config.custom_deploy_parameters.predict_function,
+    ]
+
+    # verify if there is an active stack before starting the service
+    if not context.stack:
```

Comment:
An stack is guaranteed to be active, because you already called `model_deployer = KServeModelDeployer.get_active_model_deployer()`, but this is probably just for mypy.